pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -355,6 +358,46 @@ type NginxIngress struct {
 	ExternalTrafficPolicy *corev1.ServiceExternalTrafficPolicyType `json:"externalTrafficPolicy,omitempty" protobuf:"bytes,3,opt,name=externalTrafficPolicy,casttype=k8s.io/api/core/v1.ServiceExternalTrafficPolicyType"`
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// ShootControlPlane relevant types                                                             //
+//////////////////////////////////////////////////////////////////////////////////////////////////
+
+// ShootControlPlane holds information about the general settings for the control plane of a shoot.
+type ShootControlPlane struct {
+	// HighAvailability holds the configuration settings for high availability of the
+	// control plane of a shoot.
+	HighAvailability HighAvailability `json:"highAvailability,omitempty" protobuf:"bytes,1,opt,name=highAvailability"`
+}
+
+// HighAvailability specifies the configuration settings for high availability of the
+// control plane of a shoot.
+type HighAvailability struct {
+	// FailureTolerance holds information about failure tolerance level of a
+	// highly available shoot control plane.
+	FailureTolerance FailureTolerance `json:"failureTolerance,omitempty" protobuf:"bytes,1,opt,name=failureTolerance"`
```

Comment:
You should not have `omitempty` tags for non-optional/required fields.

Reply:
Yes, there are mistakes in this PR. I am currently correcting them

Reply:
This has been corrected.